### DISCUSSION TITLE
What is the best approach to make a simple tap in appium java-client 8.x?

### DISCUSSION THREAD

**Discussion by MoOoG** [appium/java-client#1820] [Q&A]
I have found few articles on how to use new w3c actions for complex gestures using java-client 8, but there are no examples how to make a simple tap on X and Y coordinate (with a duration of a click)? I'm looking for support on iOS and Android.

**Comment by Higgens75** [comment]
you mean actionChains ?
That looks something like this (thats for click on point position)

PointerInput finger = new PointerInput(org.openqa.selenium.interactions.PointerInput.Kind.TOUCH, "finger");
org.openqa.selenium.interactions.Sequence clickPos = new org.openqa.selenium.interactions.Sequence(finger, 1);
clickPos
.addAction(finger.createPointerMove(Duration.ZERO, Origin.viewport(), pointA_X,pointA_Y))
.addAction(finger.createPointerDown(MouseButton.LEFT.asArg()))
.addAction(finger.createPointerUp(MouseButton.LEFT.asArg()));
driver.perform(Arrays.asList(clickPos));

**Reply by Higgens75** [reply]
for longpress like that

.addAction(finger.createPointerMove(Duration.ZERO, Origin.viewport(), pointA_X,pointA_Y))
.addAction(finger.createPointerDown(MouseButton.LEFT.asArg()))
.addAction(finger.createPointerMove(Duration.ofMillis(pressMillis), Origin.viewport(), pointA_X,pointA_Y))
.addAction(finger.createPointerUp(MouseButton.LEFT.asArg()));
